gateway/ln-gateway/src/rpc/mod.rs

```diff
@@ -50,11 +51,32 @@ pub struct DepositAddressPayload {
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct WithdrawPayload {
     pub federation_id: FederationId,
-    #[serde(with = "bitcoin::util::amount::serde::as_sat")]
-    pub amount: bitcoin::Amount,
+    pub amount: BitcoinAmountOrAll,
     pub address: Address,
 }
 
+/// Amount of bitcoin to send, or "all" to send all available funds
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum BitcoinAmountOrAll {
+    Amount(#[serde(with = "bitcoin::util::amount::serde::as_sat")] bitcoin::Amount),
+    All,
+}
```

Comment:
Do we want the serialization of `All` to be `null`? https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=50dfc340d46f6d2652902b19a56ff6c1